azure/services/virtualnetworks/virtualnetworks_test.go

```diff
@@ -18,322 +18,94 @@ package virtualnetworks
 
 import (
 	"context"
-	"net/http"
+	"errors"
 	"testing"
 
-	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/network/armnetwork/v4"
-	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/resources/armresources"
-	"github.com/Azure/go-autorest/autorest"
+	asonetworkv1 "github.com/Azure/azure-service-operator/v2/api/network/v1api20201101"
+	"github.com/Azure/azure-service-operator/v2/pkg/common/labels"
+	"github.com/Azure/azure-service-operator/v2/pkg/genruntime"
 	. "github.com/onsi/gomega"
 	"go.uber.org/mock/gomock"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/utils/ptr"
 	infrav1 "sigs.k8s.io/cluster-api-provider-azure/api/v1beta1"
-	"sigs.k8s.io/cluster-api-provider-azure/azure"
-	"sigs.k8s.io/cluster-api-provider-azure/azure/services/async/mock_async"
 	"sigs.k8s.io/cluster-api-provider-azure/azure/services/virtualnetworks/mock_virtualnetworks"
-	gomockinternal "sigs.k8s.io/cluster-api-provider-azure/internal/test/matchers/gomock"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	fakeclient "sigs.k8s.io/controller-runtime/pkg/client/fake"
 )
 
-var (
-	fakeVNetSpec = VNetSpec{
-		ResourceGroup:  "test-group",
-		Name:           "test-vnet",
-		CIDRs:          []string{"10.0.0.0/8"},
-		Location:       "test-location",
-		ClusterName:    "test-cluster",
-		AdditionalTags: map[string]string{"foo": "bar"},
-	}
-
-	managedTags = armresources.TagsResource{
-		Properties: &armresources.Tags{
-			Tags: map[string]*string{
-				"foo": ptr.To("bar"),
-				"sigs.k8s.io_cluster-api-provider-azure_cluster_test-cluster": ptr.To("owned"),
+func TestPostCreateOrUpdateResourceHook(t *testing.T) {
+	t.Run("failed to create or update", func(t *testing.T) {
```

Comment:
Q: Do you think it's worth it to set up a `testcases` struct like before here? I think it's probably fine now since there's only two test cases, but in the future if we have further logic after creating or updating, we might need to refactor.

Reply:
I'd vote to cross that bridge when we get there.